docker/pypi/wmagent/init.sh

```diff
@@ -98,54 +101,60 @@ basic_checks() {
-    #       * creation of all config directories if missing at the mount point
-    #          * reimplement init_install_dir
-    #          * reimplement init_config_dir
-    #       * copy/override the manage file at the host mount point with the manage file from the image deployment area
-    #       * copy/override all config files if the agent have never been initialised
-    #       * create/touch a .dockerInit file containing the WMA_BUILD_ID of the current docker image
-    #         * eventually the docker container Id may be considered in the future as well (the unique hash id to be used not the contaner name)
-    #
-    # NOTE: On every step we need to check the .dockerInit file content. There are two level of comparision we can make:
-    #       * the current container Id with the already intialised one: if we want reinitailisation on every container kill/start
-    #       * the current image Id with the already initialised one: if we want reinitialisation only on docker image rebuild (New WMAgent deployment).
-    #       THE implementation considers the later - reinitialisation on container rebuild
-    local stepMsg="Performing Docker image to Host initialisation steps"
+    # This function does all Host modifications needed at Runtime
+
+    local stepMsg="Performing Host initialization steps"
     echo "-------------------------------------------------------"
     echo "Start: $stepMsg"
 
-    # TODO: This is to be removed once we decide to run it directly from the deploy area
-    echo "$FUNCNAME: Copy the proper manage file"
-    cp -fv $WMA_DEPLOY_DIR/bin/manage $WMA_MANAGE_DIR/manage && chmod 755 $WMA_MANAGE_DIR/manage
+    # TODO: This is to be removed once we decide to run it only from the deploy area
+    #       The only place left where the manage script is called directly from
+    #       $WMA_CONFIG_DIR is from $WMA_DEPLOY_DIR/bin/clean-oracle
+    echo "$FUNCNAME: Linking the proper manage file from Config Area"
+
+    if [[ -h $WMA_CONFIG_DIR/manage ]] || [[ -f $WMA_CONFIG_DIR/manage ]]; then
+        rm -f $WMA_CONFIG_DIR/manage
+    fi
+    ln -s $WMA_MANAGE_DIR/manage $WMA_CONFIG_DIR/manage
 
     echo "$FUNCNAME: Copy the Runtime scripts"
-    _init_valid $wmaInitRuntime || {
-        # checking if $WMA_DEPLOY_DIR is root path for $pythonLib:
-        if [[ $pythonLib =~ ^$WMA_DEPLOY_DIR ]]; then
-            mkdir -p $WMA_INSTALL_DIR/Docker/
-            cp -rav $pythonLib/WMCore/WMRuntime $WMA_INSTALL_DIR/Docker/
-            cp -rav $WMA_DEPLOY_DIR/etc/ $WMA_CONFIG_DIR/
-            echo $WMA_BUILD_ID > $wmaInitRuntime
-        else
-            echo "$FUNCNAME: ERROR: \$WMA_DEPLOY_DIR: $WMA_DEPLOY_DIR is not a root path for \$pythonLib: $pithonLib"
-            echo "$FUNCNAME: ERROR: We cannot find the correct WMCore/WMRuntime source to copy at the current host!"
-            return $(false)
-        fi
-    }
+    _init_valid $wmaInitRuntime || _copy_runtime
```

Comment:
This behavior is not clear to me. If we stop a container running 2.3.3.1 and start a container 2.3.3.2, will the runtime code be copied over once again?

From what I understood, this `$wmaInitRuntime` contains the WMAgent hash build id based on the release cycle. Given that the release cycle is the same, it will not retrigger the runtime copy when upgrading a patch release. Can you please correct/confirm this understanding?

If it does not copy, then we will have to change this behavior. As a patch release might bring changes to the WMRuntime package and we need to ensure that those are properly deployed together with a new docker image version.

Reply:
> This behavior is not clear to me. If we stop a container running 2.3.3.1 and start a container 2.3.3.2, will the  runtime code be copied over once again?

Not any more

> From what I understood, this $wmaInitRuntime contains the WMAgent hash build id based on the release cycle. > Given that the release cycle is the same, it will not retrigger the runtime copy when upgrading a patch release.  Can you please correct/confirm this understanding?

That is correct. It won't re-trigger the runtime code copy step.

> If it does not copy, then we will have to change this behavior.

That's  going to be tricky, because it is not about to go through the initialization steps at all on a patch version or release candidate change, so the process would never land here. I'll have to think of something.

Reply:
ok, I think I have an idea on how to fix this behavior, but I really want to stress ... this is something we should have caught while resolving this issue: https://github.com/dmwm/WMCore/issues/11990 and the change which will follow does not relate in any way to Tier0. I am just going to use this PR to fix it.

Reply:
ok ... the intervention would be too intrusive and we better do it separately. I'll reopen the other issue and will follow with a new PR just for this.

Reply:
Yes, it's probably a better idea. Thanks